corpus/convert.py

```diff
@@ -326,3 +330,69 @@ def run(self):
                     segment.fullname() in segments_whitelist
                 ):
                     f.write(segment.orth + "\n")
+
+
+class CorpusToRawWavHDFJob(Job):
+    """
+    Gets audio files from a Bliss corpus and stores them as HDF file
+    compatible with the returnn HDFDataset
+    """
+
+    def __init__(
+        self,
+        bliss_corpus: tk.Path,
+        segment_file: Optional[tk.Path] = None,
+        returnn_root: Optional[tk.Path] = None,
+    ):
+        """
+
+        :param bliss_corpus: Bliss corpus to read segments and audio files from
+        :param segment_file: segment file that lists allowed segments
+        :param returnn_root: returnn repository
+        """
+        self.set_vis_name("Dump audio to HDF")
+
+        self.bliss_corpus = bliss_corpus
+        self.segment_file = segment_file
+        self.returnn_root = returnn_root
+
+        self.out_hdf = self.output_path("audio.hdf")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        SimpleHDFWriter = get_returnn_simple_hdf_writer(self.returnn_root.get_path())
+
+        c = corpus.Corpus()
+        c.load(self.bliss_corpus.get_path())
+
+        if self.segment_file:
+            with uopen(self.segment_file, "rt") as f:
+                segments_whitelist = set(
+                    l.strip() for l in f.readlines() if len(l.strip()) > 0
+                )
+        else:
+            segments_whitelist = None
+
+        out_hdf = SimpleHDFWriter(filename=self.out_hdf, dim=1)
+
+        for segment in c.segments():
+            if (not segments_whitelist) or (segment.fullname() in segments_whitelist):
+                audio_file = segment.recording.audio
+                audio = wave.open(audio_file)
+                audio.setpos(int(segment.start * audio.getframerate()))
+                data = audio.readframes(
+                    int((segment.end - segment.start) * audio.getframerate())
+                )
+                audio.close()
+                data_s16 = numpy.frombuffer(data, dtype=numpy.int16)
+                data_f32 = data_s16.astype(numpy.float32)
```

Comment:
We always have features as float. but storing as signed integer would save 50% of storage space. 

Are integer features supported on returnn side?
Do we want to save space on disk and do the conversion in the network?

Reply:
It seems RETURNN is just passing the data type as is without modification. So if you store numpy.int16 it will feed it this way to the tensorflow graph, mostly like resulting in an error. I already discussed with @christophmluscher that we might want to have a "pre_process" parameter for the HDF Dataset just like for the OggZip.

Reply:
what if the first layer in the feature extraction would to the casting?

Reply:
"Probably" fine...

Reply:
@JackTemaki and I discussed offline and came to the conclusion that we want to have s16 as the default data type (and conversion to float would be done inside the network) to save storage space.
If someone wanted to have float, there is a parameter to override it.